include/ginkgo/core/base/lin_op.hpp

```diff
@@ -469,12 +469,44 @@ class Permutable {
 public:
     virtual ~Permutable() = default;
 
+    /**
+     * Returns a LinOp representing the symmetric row and column permutation of
+     * the Permutable object.
+     *
+     * @param permutation_indices  the array of indices contaning the
+     *                             permutation order.
+     *
+     * @return a pointer to the new permuted object
+     */
+    virtual std::unique_ptr<LinOp> permute(
+        const Array<IndexType> *permutation_indices) const
+    {
+        return as<Permutable>(this->row_permute(permutation_indices))
+            ->column_permute(permutation_indices);
+    };
+
+    /**
+     * Returns a LinOp representing the symmetric inverse row and column
+     * permutation of the Permutable object.
+     *
+     * @param permutation_indices  the array of indices contaning the
+     *                             permutation order.
+     *
+     * @return a pointer to the new permuted object
+     */
+    virtual std::unique_ptr<LinOp> inverse_permute(
+        const Array<IndexType> *permutation_indices) const
+    {
+        return as<Permutable>(this->inverse_row_permute(permutation_indices))
+            ->inverse_column_permute(permutation_indices);
+    };
+
     /**
      * Returns a LinOp representing the row permutation of the Permutable
      * object.
      *
      * @param permutation_indices  the array of indices contaning the
```

Comment:
```suggestion
     * @param permutation_indices  the array of indices containing the
```
also applying to other contaning